Entropy criterion is broken

Looks like we accidentally started relying on criteria-specific parameter somewhere outside of the criteria code. Running `nvbench.example.cpp20.auto_throughput --stopping-criterion entropy` leads to:
```bash
Fail: Unexpected error: /home/coder/nvbench/nvbench/named_values.cxx:113: Error looking up float64 value `min-time`:
```

--- a/nvbench/detail/measure_cold.cu
+++ b/nvbench/detail/measure_cold.cu
@@ -401,7 +401,7 @@ void measure_cold_base::generate_summaries()
       get_param(max_noise, "max-noise");
 
       std::optional<nvbench::float64_t> min_time;
-      get_param(max_noise, "min-time");
+      get_param(min_time, "min-time");
 
       if (max_noise && cuda_noise > *max_noise)
       {

--- a/nvbench/detail/measure_cpu_only.cxx
+++ b/nvbench/detail/measure_cpu_only.cxx
@@ -210,11 +210,22 @@ void measure_cpu_only_base::generate_summaries()
 
     if (m_max_time_exceeded)
     {
-      const auto timeout   = m_walltime_timer.get_duration();
-      const auto max_noise = m_criterion_params.get_float64("max-noise");
-      const auto min_time  = m_criterion_params.get_float64("min-time");
+      const auto timeout = m_walltime_timer.get_duration();
 
-      if (cpu_noise > max_noise)
+      auto get_param = [this](std::optional<nvbench::float64_t> &param, const std::string &name) {
+        if (m_criterion_params.has_value(name))
+        {
+          param = m_criterion_params.get_float64(name);
+        }
+      };
+
+      std::optional<nvbench::float64_t> max_noise;
+      get_param(max_noise, "max-noise");
+
+      std::optional<nvbench::float64_t> min_time;
+      get_param(min_time, "min-time");
+
+      if (max_noise && cpu_noise > *max_noise)
       {
         printer.log(nvbench::log_level::warn,
                     fmt::format("Current measurement timed out ({:0.2f}s) "
@@ -222,7 +233,7 @@ void measure_cpu_only_base::generate_summaries()
                                 "{:0.2f}%)",
                                 timeout,
                                 cpu_noise * 100,
-                                max_noise * 100));
+                                *max_noise * 100));
       }
       if (m_total_samples < m_min_samples)
       {
@@ -233,7 +244,7 @@ void measure_cpu_only_base::generate_summaries()
                                 m_total_samples,
                                 m_min_samples));
       }
-      if (m_total_cpu_time < min_time)
+      if (min_time && m_total_cpu_time < *min_time)
       {
         printer.log(nvbench::log_level::warn,
                     fmt::format("Current measurement timed out ({:0.2f}s) "
@@ -241,7 +252,7 @@ void measure_cpu_only_base::generate_summaries()
                                 "{:0.2f}s)",
                                 timeout,
                                 m_total_cpu_time,
-                                min_time));
+                                *min_time));
       }
     }
 

--- a/nvbench/state.cuh
+++ b/nvbench/state.cuh
@@ -174,7 +174,11 @@ struct state
   /// Only applies to `stdrel` stopping criterion. @{
   [[nodiscard]] nvbench::float64_t get_min_time() const
   {
-    return m_criterion_params.get_float64("min-time");
+    if (m_criterion_params.has_value("min-time"))
+    {
+      return m_criterion_params.get_float64("min-time");
+    }
+    return 0.;
   }
   void set_min_time(nvbench::float64_t min_time)
   {
@@ -188,7 +192,11 @@ struct state
   /// Only applies to `stdrel` stopping criterion. @{
   [[nodiscard]] nvbench::float64_t get_max_noise() const
   {
-    return m_criterion_params.get_float64("max-noise");
+    if (m_criterion_params.has_value("max-noise"))
+    {
+      return m_criterion_params.get_float64("max-noise");
+    }
+    return 1.;
   }
   void set_max_noise(nvbench::float64_t max_noise)
   {

